scripts/variantstore/wdl/MergePgenHierarchical.wdl

```diff
@@ -0,0 +1,318 @@
+            fi
+        done
+
+        case $count in
+        0)
+            echo "No pgen files so creating empty ones"
+            touch ~{output_file_base_name}.pgen
+            touch ~{output_file_base_name}.pvar
+            touch ~{output_file_base_name}.psam
+            ;;
+        1)
+            echo "Only one pgen file so renaming the files for output"
+            pgen_basename=$(cat mergelist.txt | xargs)
+            mv ${pgen_basename}.pgen ~{output_file_base_name}.pgen
+            mv ${pgen_basename}.psam ~{output_file_base_name}.psam
+            if test -f ${pgen_basename}.pvar
+            then
+                mv ${pgen_basename}.pvar ~{output_file_base_name}.pvar
+            else
+                mv ${pgen_basename}.pvar.zst ~{output_file_base_name}.pvar.zst
+            fi
+            ;;
+        *)
+            echo "${count} pgen files, merging"
+            if head -n 1 "~{pvar_list}" | grep -q "pvar$"
+            then
+                plink2 --pmerge-list mergelist.txt --memory 10000 --threads ~{threads} --out ~{output_file_base_name} --pmerge-output-vzs
+            else
+                plink2 --pmerge-list mergelist.txt pfile-vzs --memory 10000 --threads ~{threads} --out ~{output_file_base_name} --pmerge-output-vzs
+            fi
+            ;;
+        esac
+
+    >>>
+
+    output {
+        File pgen_file = "${output_file_base_name}.pgen"
+        File pvar_file = "${output_file_base_name}.pvar.zst"
+        File psam_file = "${output_file_base_name}.psam"
+    }
+
+    runtime {
+        docker: "${plink_docker}"
+        memory: "12 GB"
+        disks: "local-disk ${disk_in_gb} HDD"
+        bootDiskSizeGb: 15
+        cpu: "${cpu}"
+    }
+}
+
+task MakeFileLists {
+    input {
+        Array[File] pgen_files
+        Array[File] pvar_files
+        Array[File] psam_files
+    }
+
+    parameter_meta {
+        pgen_files: {
+            localization_optional: true
+        }
+        pvar_files: {
+            localization_optional: true
+        }
+        psam_files: {
+            localization_optional: true
+        }
+    }
+    
+    meta {
+        # This causes issues when call cached for some reason, so we don't want to do that
+        volatile: true
+    }
+
+    command <<<
+        set -euxo pipefail
+
+        touch pgen_list.txt
+        touch psam_list.txt
+        touch pvar_list.txt
+
+        PGEN_ARRAY=(~{sep=" " pgen_files})
+        PSAM_ARRAY=(~{sep=" " psam_files})
+        PVAR_ARRAY=(~{sep=" " pvar_files})
+
+        for i in "${!PGEN_ARRAY[@]}"
+        do
+            echo "${PGEN_ARRAY[$i]}" >> pgen_list.txt
+            echo "${PSAM_ARRAY[$i]}" >> psam_list.txt
+            echo "${PVAR_ARRAY[$i]}" >> pvar_list.txt
+        done
+    >>>
+
+    output {
+        File pgen_list = "pgen_list.txt"
+        File psam_list = "psam_list.txt"
+        File pvar_list = "pvar_list.txt"
+    }
+
+    runtime {
+        docker: "ubuntu:22.04"
```

Comment:
Is this code very particular about being pinned to this version of Ubuntu or could we specify `basic_docker` as defined by `GvsUtils.wdl`?

Reply:
I can make that change.  One or two of the tasks in that file need that specific docker because of some bash syntax I'm using, but I'm pretty sure that task does not.